influxdata/value.go

```diff
@@ -126,44 +126,89 @@ func NewValueFromBytes(kind ValueKind, data []byte) (Value, error) {
 func NewValue(x interface{}) (Value, bool) {
 	switch x := x.(type) {
 	case int64:
-		return Value{
-			number: uint64(x),
-			bytes:  intSentinel[:],
-		}, true
+		return IntValue(x), true
 	case uint64:
-		return Value{
-			number: uint64(x),
-			bytes:  uintSentinel[:],
-		}, true
+		return UintValue(x), true
 	case float64:
-		if math.IsInf(x, 0) || math.IsNaN(x) {
-			return Value{}, false
-		}
-		return Value{
-			number: math.Float64bits(x),
-			bytes:  floatSentinel[:],
-		}, true
+		return FloatValue(x)
 	case bool:
-		n := uint64(0)
-		if x {
-			n = 1
-		}
-		return Value{
-			number: uint64(n),
-			bytes:  boolSentinel[:],
-		}, true
+		return BoolValue(x), true
 	case string:
-		return Value{
-			bytes: []byte(x),
-		}, true
+		return StringValue(x)
 	case []byte:
-		return Value{
-			bytes: append([]byte(nil), x...),
-		}, true
+		return StringValueFromBytes(x)
 	}
 	return Value{}, false
 }
 
+// IntValue returns a Value containing the value of x.
+func IntValue(x int64) Value {
+	return Value{
+		number: uint64(x),
+		bytes:  intSentinel[:],
+	}
+}
+
+// UintValue returns a Value containing the value of x.
+func UintValue(x uint64) Value {
+	return Value{
+		number: uint64(x),
+		bytes:  uintSentinel[:],
+	}
+}
+
+// FloatValue returns a Value containing the value of x.
+//
+// FloatValue will fail and return false if x is a +/- infinity or NaN.
```

Comment:
```suggestion
// FloatValue will fail and return false if x is non-finite.
```

? (that's the terminology we use elsewhere in the godoc)